core/src/main/kotlin/no/strooped/view/screen/LobbyScreen.kt

```diff
@@ -0,0 +1,145 @@
+    private val refreshInterval = 0.1f
+
+    init {
+        cam.setToOrtho(false, Gdx.graphics.width.toFloat(), Gdx.graphics.height.toFloat())
+        batch = SpriteBatch()
+
+        loadAnim = arrayOfNulls(numberOfFrames)
+        for (i in 0 until numberOfFrames) {
+            val nameOfImage = "load$i.jpg"
+            loadAnim[i] = Texture(nameOfImage)
+        }
+        val messsage = "Hey " + controller.username + "!\nPlease wait for the lobby-master to start " +
+            "the game.\nWe made a spinning circle you can enjoy while you wait."
+        val label = Label(messsage, Label.LabelStyle(BitmapFont(Gdx.files.internal("chunkfive.fnt")), Color.FIREBRICK))
+        label.setFontScale(FONT_SIZE_WELCOME_TEXT)
+        val labelWidth = Gdx.graphics.width * 0.8f
+        label.width = labelWidth
+        val labelPosX = (Gdx.graphics.width - label.width) * 0.5f
+        val labelPosY = Gdx.graphics.height * 0.7f
+        label.setPosition(labelPosX, labelPosY)
+        label.setWrap(true) // fit inside the label
+        label.setAlignment(1) // center the text
+        ui.addActor(label)
+
+        /*val toastFactory: Toast.ToastFactory = Builder()
+            .font(attr.font)
+            .build()*/
+        // https://github.com/wentsa/Toast-LibGDX - for toasts
+    }
+
+    override fun hide() {}
+
+    override fun show() {
+        batch = SpriteBatch()
+    }
+
+    override fun render(delta: Float) {
+        handleInput()
+        elapsedTime += Gdx.graphics.deltaTime
+        batch.projectionMatrix = cam.combined
+        batch.begin()
+        batch.draw(
+            background,
+            backPosX,
+            backPosY,
+            Gdx.graphics.width.toFloat(),
+            Gdx.graphics.height.toFloat()
+        )
+        batch.draw(
+            exitBtn,
+            exitBtnPosX,
+            exitBtnPosY,
+            TextureSizes.exitGameButtonWidth(),
+            TextureSizes.exitGameButtonHeight()
+        )
+        if (elapsedTime > refreshInterval) {
+            iterat++
+            elapsedTime = 0f
+            if (iterat == numberOfFrames) {
+                iterat = 0
+            }
+        }
+        batch.draw(
+            loadAnim[iterat],
+            loadPosX,
+            loadPosY,
+            TextureSizes.loadWidth(),
+            TextureSizes.loadHeight())
+        batch.end()
+        ui.draw()
+    }
+
+    override fun pause() {}
+
+    override fun resume() {}
+
+    override fun resize(width: Int, height: Int) {
+        ui.viewport?.update(width, height)
+    }
+
+    override fun dispose() {
+        ui.dispose()
+        batch.dispose()
+    }
+
+    private fun handleInput() {
+        if (Gdx.input.justTouched()) {
+            val rec = Rectangle(
+                exitBtnPosX,
+                exitBtnPosY,
+                TextureSizes.exitGameButtonWidth(),
+                TextureSizes.exitGameButtonHeight())
+            val touchX = Gdx.input.x.toFloat()
```

Comment:
Same comment here as in the JoinGameScreen. I think we will save ourselves alot of additional code, 
if we have a cusotm button, with an onClick-method. Then we can make use of kotlins strength (higher-order-programming and anonymous functions)

```kotlin
// Example of how the interface could look like
exitButton.onClick { 
  controller.exitLobby()
}
```

Reply:
As I said, I'll take a look at it

Reply:
Done